### PR TITLE
EKF2: origin reset cleanup

### PR DESCRIPTION
**Pull request by bresch**
We now have 2 options to set the origin of the local frame:
- set the origin given the lat/lon/alt of the origin
- set the origin given the current lat/lon/alt of the vehicle

### CODE REVIEW

File: src/modules/ekf2/EKF/ekf.cpp [hunk 1]
```diff
@@ -280,8 +280,7 @@ bool Ekf::resetGlobalPosToExternalObservation(double lat_deg, double lon_deg, fl
 		uint64_t timestamp_observation)
 {
 	if (!_pos_ref.isInitialized()) {
-		ECL_WARN("unable to reset global position, position reference not initialized");
-		return false;
+		return setEkfGlobalOriginFromCurrentPos(lat_deg, lon_deg, NAN, accuracy, NAN);
```
Comment (dagar) [inline src/modules/ekf2/EKF/ekf.cpp:283]: A signature like this is a bit awkward when you only have part of the data, why not use the helper for what you actually have? Then `setEkfGlobalOriginFromCurrentPos()` could actually succeed or fail atomically for all of lat/lon/alt.

```suggestion
		return setLatLonOriginFromCurrentPos(lat_deg, lon_deg, accuracy);
```

**Comment by hamishwillee** [comment]
@bresch I thought the EKF2 origin was effectively the position of the IMU on boot. Does this change anything for docs/user-facing behaviour?

**Comment by bresch** [comment]
> I thought the EKF2 origin was effectively the position of the IMU on boot. Does this change anything for docs/user-facing behaviour?

@hamishwillee By default it is, yes, but the user can change it by sending a [mavlink command](https://mavlink.io/en/messages/common.html#SET_GPS_GLOBAL_ORIGIN).

File: src/modules/ekf2/EKF/ekf.h [hunk 2]
```diff
@@ -258,7 +258,11 @@ class Ekf final : public EstimatorInterface
 	// get the ekf WGS-84 origin position and height and the system time it was last set
 	// return true if the origin is valid
 	bool getEkfGlobalOrigin(uint64_t &origin_time, double &latitude, double &longitude, float &origin_alt) const;
-	bool setEkfGlobalOrigin(double latitude, double longitude, float altitude, float eph = 0.f, float epv = 0.f);
+	bool checkLatLonValidity(double latitude, double longitude);
+	bool checkAltitudeValidity(float altitude);
+	bool setEkfGlobalOrigin(double latitude, double longitude, float altitude, float eph = NAN, float epv = NAN);
+	bool setEkfGlobalOriginFromCurrentPos(double latitude, double longitude, float altitude, float eph = NAN,
+					      float epv = 0.f);
```
Comment (bresch) [inline src/modules/ekf2/EKF/ekf.h:265]:
```suggestion
					      float epv = NAN);
```

File: src/modules/ekf2/EKF/ekf.h [hunk 3]
```diff
@@ -758,6 +762,12 @@ class Ekf final : public EstimatorInterface
 		P.slice<S.dof, S.dof>(S.idx, S.idx) = cov;
 	}
 
+	bool setLatLonOrigin(double latitude, double longitude, float eph);
```
Comment (bresch) [inline src/modules/ekf2/EKF/ekf.h:765]:
```suggestion
	bool setLatLonOrigin(double latitude, double longitude, float eph = NAN);
```

**Pull request merged**